File: Esame3Java/app/src/main/java/com/example/esame3java/SecondActivity.java
package com.example.esame3java;

import android.app.Activity;
import android.database.Cursor;
import android.os.Bundle;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class SecondActivity extends Activity {

    ListView lvLeaderboard;
    CustomAdapter adapter;

    List<Score> top3 = new ArrayList<>();
    DatabaseHelper dbHelper = new DatabaseHelper(this);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_secondary);

        Cursor cursor = dbHelper.getAllScores();
        if (cursor != null && cursor.moveToFirst()) {
            do {
                String name = cursor.getString(cursor.getColumnIndexOrThrow(DatabaseHelper.COLUMN_NAME));
                int score = cursor.getInt(cursor.getColumnIndexOrThrow(DatabaseHelper.COLUMN_SCORE));
                top3.add(new Score(score, name));
            } while (cursor.moveToNext());
            cursor.close();
            top3.sort(Comparator.comparingInt(Score::getScore).reversed());
        }

        lvLeaderboard = findViewById(R.id.lvLeaderboard);
        adapter = new CustomAdapter(this, R.layout.score_element, top3);
        lvLeaderboard.setAdapter(adapter);
    }
}

File: Esame3Kotlin/app/src/main/java/com/example/esame1/SecondActivity.kt
package com.example.esame1

import android.os.Bundle
import android.widget.Adapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.ListAdapter


class SecondActivity: AppCompatActivity() {

    lateinit var btnInsert: Button
    lateinit var etName: EditText
    lateinit var dbHelper: DatabaseHelper
    lateinit var scores: MutableList<Score>
    lateinit var lvScoreBoard : ListView
    

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)


        dbHelper = DatabaseHelper(this)
        btnInsert = findViewById(R.id.btnInsert)
        etName = findViewById(R.id.etName)
        lvScoreBoard = findViewById(R.id.lvScoreboard)
        val curScore : Int = intent.getIntExtra("curScore", 0)
        uploadSB()

         btnInsert.setOnClickListener{
             dbHelper.addScore((etName.text).toString(), curScore )
             uploadSB()
         }


    }

    fun uploadSB(){
        scores = dbHelper.getAllScores().toMutableList()
        scores = scores.sortedByDescending { it.score } as MutableList<Score>
        val adapter = ScoreAdapter(this, scores.take(3))
        lvScoreBoard.adapter = adapter
    }
}